tests/test_blueprints.py

```diff
@@ -1112,3 +1112,21 @@ async def index(_):
     app.router.finalize()
 
     assert app.router.routes[0].path == "foo/"
+
+
+def test_bp_allow_override(app: Sanic):
+    bp = Blueprint("test_text", allow_route_overwrite=True)
+
+    @bp.route("/")
+    def handler1(request):
+        return text("Hello")
+
+    @bp.route("/")
+    def handler2(request):
+        return text("Overwritten Hello")
+
+    app.blueprint(bp)
+    _, response = app.test_client.get("/")
+
```

Comment:
This pattern strikes me as dangerous. What would the use case be when we would want to allow this? On `copy` it makes sense, but why here?

Reply:
Because the flag in at the `Blueprint` object attribute level so it allows this when the flag is set to `True` when the object created. If this is not desired and looks dangerous, I can remove the flag parameter from the `Blueprint` object constructor to make it safer. How does that sound?

Reply:
Yeah, I think we should not expose it. If if needs to be a flag on the Blueprint, let's make it internal `_allow_route_overwrite` and not expose it "publicly".

Reply:
Updated